examples_tests/61.UI/UI_System.cpp

```diff
@@ -0,0 +1,1102 @@
+		0x0007000f,0x00000004,0x00000004,0x6e69616d,0x00000000,0x00000009,0x0000000d,0x00030010,
+		0x00000004,0x00000007,0x00030003,0x00000002,0x000001c2,0x00040005,0x00000004,0x6e69616d,
+		0x00000000,0x00040005,0x00000009,0x6c6f4366,0x0000726f,0x00030005,0x0000000b,0x00000000,
+		0x00050006,0x0000000b,0x00000000,0x6f6c6f43,0x00000072,0x00040006,0x0000000b,0x00000001,
+		0x00005655,0x00030005,0x0000000d,0x00006e49,0x00050005,0x00000016,0x78655473,0x65727574,
+		0x00000000,0x00040047,0x00000009,0x0000001e,0x00000000,0x00040047,0x0000000d,0x0000001e,
+		0x00000000,0x00040047,0x00000016,0x00000022,0x00000000,0x00040047,0x00000016,0x00000021,
+		0x00000000,0x00020013,0x00000002,0x00030021,0x00000003,0x00000002,0x00030016,0x00000006,
+		0x00000020,0x00040017,0x00000007,0x00000006,0x00000004,0x00040020,0x00000008,0x00000003,
+		0x00000007,0x0004003b,0x00000008,0x00000009,0x00000003,0x00040017,0x0000000a,0x00000006,
+		0x00000002,0x0004001e,0x0000000b,0x00000007,0x0000000a,0x00040020,0x0000000c,0x00000001,
+		0x0000000b,0x0004003b,0x0000000c,0x0000000d,0x00000001,0x00040015,0x0000000e,0x00000020,
+		0x00000001,0x0004002b,0x0000000e,0x0000000f,0x00000000,0x00040020,0x00000010,0x00000001,
+		0x00000007,0x00090019,0x00000013,0x00000006,0x00000001,0x00000000,0x00000000,0x00000000,
+		0x00000001,0x00000000,0x0003001b,0x00000014,0x00000013,0x00040020,0x00000015,0x00000000,
+		0x00000014,0x0004003b,0x00000015,0x00000016,0x00000000,0x0004002b,0x0000000e,0x00000018,
+		0x00000001,0x00040020,0x00000019,0x00000001,0x0000000a,0x00050036,0x00000002,0x00000004,
+		0x00000000,0x00000003,0x000200f8,0x00000005,0x00050041,0x00000010,0x00000011,0x0000000d,
+		0x0000000f,0x0004003d,0x00000007,0x00000012,0x00000011,0x0004003d,0x00000014,0x00000017,
+		0x00000016,0x00050041,0x00000019,0x0000001a,0x0000000d,0x00000018,0x0004003d,0x0000000a,
+		0x0000001b,0x0000001a,0x00050057,0x00000007,0x0000001c,0x00000017,0x0000001b,0x00050085,
+		0x00000007,0x0000001d,0x00000012,0x0000001c,0x0003003e,0x00000009,0x0000001d,0x000100fd,
+		0x00010038
+	};
+
+	struct State
+	{
+		smart_refctd_ptr<ILogicalDevice> device{};
+		smart_refctd_ptr<IGPUSampler> fontSampler{};
+		smart_refctd_ptr<IDescriptorPool> descriptorPool{};
+		smart_refctd_ptr<IGPUDescriptorSet> gpuDescriptorSet{};
+		smart_refctd_ptr<IGPURenderpassIndependentPipeline> independentPipeline{};
+		smart_refctd_ptr<IGPUGraphicsPipeline> pipeline{};
+		smart_refctd_ptr<IGPUImageView> fontTexture{};
+		smart_refctd_ptr<CommonAPI::InputSystem> inputSystem{};
+		smart_refctd_ptr<IWindow> window{};
+		std::vector<smart_refctd_ptr<IGPUBuffer>> vertexBuffers{};
+		std::vector<smart_refctd_ptr<IGPUBuffer>> indexBuffers{};
+		bool hasFocus = false;
+
+		// TODO: Use a signal class instead.//Signal<> UIRecordSignal{};
+		struct Subscriber {
+			int id = -1;
+			std::function<void()> listener = nullptr;
+		};
+		std::vector<Subscriber> subscribers{};
+	};
+
+	static State* state = nullptr;
```

Comment:
Globals are a big no-no at DevSH Graphics Programming, please refactor

Reply:
I rewrote it in a class .